x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_custom_assets_extension.tsx

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  CustomAssetsAccordionProps,
+  CustomAssetsAccordion,
+  PackageAssetsComponent,
+} from '../../../../../../../fleet/public';
+
+export const EndpointCustomAssetsExtension: PackageAssetsComponent = () => {
+  const views: CustomAssetsAccordionProps['views'] = [
+    {
+      name: 'Hosts',
+      url: '/app/security/hosts/allHosts',
```

Comment:
this goes to 404 not found as it doesn't have basepath prepended to it

Reply:
I'm trying to prepend the basepath in Custom Logs via our `startServices` hook in Fleet, but I'm getting a minified React error when trying to call any hooks from the UI extension. 

e.g. 

```tsx
export const LazyCustomLogsAssetsComponent = lazy<PackageAssetsComponent>(async () => {
  const { http } = useStartServices();
  const logStreamUrl = http.basePath.prepend('/app/logs/stream');

  const views: CustomAssetsAccordionProps['views'] = [
    {
      name: 'Logs',
      url: logStreamUrl,
      description: 'View Custom logs data in Logs app',
    },
  ];
  return {
    default: () => <CustomAssetsAccordion views={views} />,
  };
});
```

This results in:

![Screen Shot 2021-06-29 at 8 44 57 AM](https://user-images.githubusercontent.com/6766512/123799516-4d739500-d8b6-11eb-930d-3b13d42c3ecd.png)

I've tried tracking this down without much luck, I think it _may_ have something to do with context providers not being wrapped around the extension properly? I was able to access a basePath via context in the Synthetics and Endpoint UI Extensions, just not this one in the Fleet/Integrations UI app. Any ideas?

Reply:
the issue here is `LazyCustomLogsAssetsComponent` is not a react component, I had a similar issue in my PR https://github.com/elastic/kibana/pull/103550 and I adopted that pattern to fix https://github.com/elastic/kibana/blob/f3ee4b645755ced38b4b0b64da2f986e69ee493e/x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_requirements_page/components/fleet_server_on_prem_instructions.tsx#L405-L417

Reply:
Thanks @nchaulet - was able to fix this.